lib/authz/permissions_test.go

```diff
@@ -236,14 +236,29 @@ func upsertLockWithPutEvent(ctx context.Context, t *testing.T, client *testClien
 	defer lockWatch.Close()
 
 	require.NoError(t, client.UpsertLock(ctx, lock))
-	select {
-	case event := <-lockWatch.Events():
-		require.Equal(t, types.OpPut, event.Type)
-		require.Empty(t, resourceDiff(lock, event.Resource))
-	case <-lockWatch.Done():
-		t.Fatalf("Watcher exited with error: %v.", lockWatch.Error())
-	case <-time.After(2 * time.Second):
-		t.Fatal("Timeout waiting for lock put.")
+
+	// Retry a few times to wait for the resource event we expect as the
+	// resource watcher can potentially return events for previously
+	// created resources as well.
+	localCtx, cancel := context.WithDeadline(ctx, time.Now().Add(2*time.Second))
+	defer cancel()
+	for {
+		select {
+		case <-localCtx.Done():
+			t.Fatalf("Timeout waiting for lock put: %v.", lock)
+		default:
+		}
+		select {
+		case event := <-lockWatch.Events():
+			require.Equal(t, types.OpPut, event.Type)
+			if resourceDiff(lock, event.Resource) == "" {
+				return
+			}
+		case <-lockWatch.Done():
+			t.Fatalf("Watcher exited with error: %v.", lockWatch.Error())
+		case <-localCtx.Done():
+			t.Fatalf("Timeout waiting for lock put: %v.", lock)
+		}
```

Comment:
Can we transform it to use `require.Eventually`? It feels easier to read

```suggestion
	require.Eventually(t, func() bool {
		select {
		case event := <-lockWatch.Events():
			return types.OpPut == event.Type && resourceDiff(lock, event.Resource) == ""
		case <-lockWatch.Done():
			return false
		}
	}, 2*time.Second, 200*time.Millisecond)
```

Reply:
Good idea, done.